src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ProvideCorrectArgumentsToFormattingMethodsTests.cs

```diff
@@ -147,134 +264,361 @@ public class C
+        {|#1:" + invocation + @"(""{0} {1} {2}"", 1, 2)|};
+        {|#2:" + invocation + @"(""{0} {1} {2}"", 1)|};
+        {|#3:" + invocation + @"(""{2} {0} {1}"", 1)|};
+        {|#4:" + invocation + @"(""{1} {0} {1}"", 1)|};
     }
 }
 ",
-            GetCSharpResultAt(8, 17),
-            GetCSharpResultAt(9, 9),
-            GetCSharpResultAt(10, 9));
+            VerifyCS.Diagnostic(ProvideCorrectArgumentsToFormattingMethodsAnalyzer.NotEnoughArgsRule).WithLocation(0),
+            VerifyCS.Diagnostic(ProvideCorrectArgumentsToFormattingMethodsAnalyzer.NotEnoughArgsRule).WithLocation(1),
+            VerifyCS.Diagnostic(ProvideCorrectArgumentsToFormattingMethodsAnalyzer.NotEnoughArgsRule).WithLocation(2),
+            VerifyCS.Diagnostic(ProvideCorrectArgumentsToFormattingMethodsAnalyzer.NotEnoughArgsRule).WithLocation(3),
+            VerifyCS.Diagnostic(ProvideCorrectArgumentsToFormattingMethodsAnalyzer.NotEnoughArgsRule).WithLocation(4));
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System
+
+Public Class C
+    Sub Method()
+        {|#0:" + invocation + @"(""{0} {1}"", 1)|}
+        {|#1:" + invocation + @"(""{0} {1} {2}"", 1, 2)|}
+        {|#2:" + invocation + @"(""{0} {1} {2}"", 1)|}
+        {|#3:" + invocation + @"(""{2} {0} {1}"", 1)|}
+        {|#4:" + invocation + @"(""{1} {0} {1}"", 1)|}
+    End Sub
+End Class",
+            VerifyVB.Diagnostic(ProvideCorrectArgumentsToFormattingMethodsAnalyzer.NotEnoughArgsRule).WithLocation(0),
+            VerifyVB.Diagnostic(ProvideCorrectArgumentsToFormattingMethodsAnalyzer.NotEnoughArgsRule).WithLocation(1),
+            VerifyVB.Diagnostic(ProvideCorrectArgumentsToFormattingMethodsAnalyzer.NotEnoughArgsRule).WithLocation(2),
+            VerifyVB.Diagnostic(ProvideCorrectArgumentsToFormattingMethodsAnalyzer.NotEnoughArgsRule).WithLocation(3),
+            VerifyVB.Diagnostic(ProvideCorrectArgumentsToFormattingMethodsAnalyzer.NotEnoughArgsRule).WithLocation(4));
         }
 
-        [Fact]
-        public async Task CA2241CSharpVarArgsNotSupported()
+        [Theory, WorkItem(1254, "https://github.com/dotnet/roslyn-analyzers/issues/1254")]
+        [InlineData("Console.Write")]
+        [InlineData("Console.WriteLine")]
+        public async Task CA2250_NotEnoughArgsMissingFormatIndexRule_ConsoleWriteMethods_Diagnostic(string invocation)
         {
-            // currently not supported due to "https://github.com/dotnet/roslyn/issues/7346"
-            await new VerifyCS.Test
-            {
-                ReferenceAssemblies = ReferenceAssemblies.NetFramework.Net472.Default,
-                TestCode = @"
+            await VerifyCS.VerifyAnalyzerAsync(@"
 using System;
 
 public class C
 {
     void Method()
     {
-        Console.Write(""{0} {1} {2} {3} {4}"", 1, 2, 3, 4, __arglist(5));
-        Console.WriteLine(""{0} {1} {2} {3} {4}"", 1, 2, 3, 4, __arglist(5));
+        {|#0:" + invocation + @"(""{1}"", 1)|};
```

Comment:
How to highlight the missing indexes?

Reply:
In _this_ case I think we want to have a diagnostic on `, 1` and say that is will not be used.

---
In reply to: [508702128](https://github.com/dotnet/roslyn-analyzers/pull/4226#discussion_r508702128) [](ancestors = 508702128)